make/targets/golang/verify-update.mk

```diff
@@ -28,6 +28,29 @@ verify-golint:
 	$(GOLINT) $(GO_PACKAGES)
 .PHONY: verify-govet
 
+verify-gomod:
+	$(info Running `$(GO) mod tidy`.)
+	@if ! [ -f go.mod ]; then \
+		echo "$@ failed - please run \`go mod init\`"; \
+		exit 1; \
+	fi; \
+	if ! [ -f go.sum ]; then \
+		echo "$@ failed - please run \`make update-gomod\`"; \
+		exit 1; \
+	fi; \
+	TMP_GOMOD=$$( mktemp ); \
+	TMP_GOSUM=$$( mktemp ); \
+	cp --preserve=all go.mod $${TMP_GOMOD}; \
+	cp --preserve=all go.sum $${TMP_GOSUM}; \
+	$(GO) mod tidy; \
+	if ! { diff $${TMP_GOMOD} go.mod && diff $${TMP_GOSUM} go.sum; }; then \
```

Comment:
are we sure a `go mod tidy` is stable through out versions and the state of the internet?

Reply:
`go mod tidy` is not supposed to reach the Internet _if_ all required dependencies are listed in go.mod and go.sum. However, it will resolve dependencies should they be missing. Is that acceptable?

Reply:
Have we verified that?

Reply:
Empyrically, yes (right now on the openshift/installer repo with go1.15.6). However I believe that there is no promise of such behaviour being consistent in future versions.

https://golang.org/cmd/go/#hdr-Add_missing_and_remove_unused_modules

Reply:
@tnozicka what do you think?